fuel-vm/src/interpreter/alu/muldiv.rs

```diff
@@ -0,0 +1,86 @@
+use super::super::{internal::inc_pc, is_wrapping, ExecutableTransaction, Interpreter};
+use crate::{constraints::reg_key::*, error::RuntimeError};
+
+use fuel_asm::PanicReason;
+use fuel_types::{RegisterId, Word};
+
+impl<S, Tx> Interpreter<S, Tx>
+where
+    Tx: ExecutableTransaction,
+{
+    /// Stores the overflowed wrapped value into RegId::OF
+    pub(crate) fn alu_muldiv(
+        &mut self,
+        ra: RegisterId,
+        lhs: Word,
+        rhs: Word,
+        divider: Word,
+    ) -> Result<(), RuntimeError> {
+        let (
+            SystemRegisters {
+                flag,
+                mut of,
+                mut err,
+                pc,
+                ..
+            },
+            mut w,
+        ) = split_registers(&mut self.registers);
+        let dest = &mut w[ra.try_into()?];
+
+        let (result, overflow) = muldiv(lhs, rhs, divider);
+
+        if overflow != 0 && !is_wrapping(flag.into()) {
+            println!("!!");
+            return Err(PanicReason::ArithmeticOverflow.into());
```

Comment:
should not panic if `F_UNSAFEMATH` is set

> If the [F_UNSAFEMATH](https://github.com/FuelLabs/fuel-vm/pull/424/index.md#flags) flag is set, an ALU operation that would have panicked will instead set $err to true.

https://github.com/FuelLabs/fuel-specs/blob/813faf0b94bc45cd5e9b55826b57831d27bed394/src/vm/instruction_set.md?plain=1#L144

Reply:
I think the spec is worded incorrectly. Unsafemath doesn't supress panic when an overflow occurs. Instead, F_WRAPPING controls this behavior. Currently F_UNSAFEMATH is only used for zero divisor, modulus and root,logarithm of zero or with a base <= 1, and (incorrectly, see https://github.com/FuelLabs/fuel-vm/issues/426) bit shifts over 2<sup>32</sup> bits.

I'll make a PR to clarify the spec. I feel like we could even combine the flags since no instruction uses both of them. We could even combine $of and $err registers into one as well. But the easiest would be to simply make the spec clear about the difference.